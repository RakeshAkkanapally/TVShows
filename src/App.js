import { makeStyles } from "@material-ui/core";
import Showpage from "./pages/Showpage";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Dashboardpage from "./pages/Dashboardpage";
import React  from "react";
import NotFoundpage from "./pages/NotFoundpage";

const useStyles = makeStyles({
  root: {
    backgroundColor: "#000",
    height: "100vh",
    overflowX: "hidden"
  }
});

function App() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Router>
        <Switch>
          <Route exact path="/" component={Dashboardpage}/>
          <Route exact path="/show/:id" component={Showpage}/>
          <Route path="*" component={NotFoundpage} />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
