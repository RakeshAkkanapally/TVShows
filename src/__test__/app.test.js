import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import ShowContext from "../reducers/showContext";
import App from "../App";

it("App renders correctly", () => {
  const tree = renderer
    .create(
      <ShowContext.Provider value={ShowContext}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </ShowContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
