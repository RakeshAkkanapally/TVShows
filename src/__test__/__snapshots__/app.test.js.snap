// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div>
    <div>
      <header
        className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-7 MuiPaper-elevation4"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
          >
            <a
              aria-current="page"
              className="makeStyles-logo-10 active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              <div
                className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 4l2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
                  />
                </svg>
                <h6
                  className="MuiTypography-root MuiTypography-h6"
                >
                  <strong>
                    TV SHOWS
                  </strong>
                </h6>
              </div>
            </a>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <div
              className="MuiPaper-root makeStyles-paper-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <a
                aria-current={null}
                href="/home"
                onClick={[Function]}
              >
                <p
                  className="MuiTypography-root makeStyles-hover-11 MuiTypography-body1"
                >
                  Search
                </p>
              </a>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <div
              className="MuiPaper-root makeStyles-paper-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <a
                aria-current="page"
                className="makeStyles-activeLink-9"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                <p
                  className="MuiTypography-root makeStyles-hover-11 MuiTypography-body1"
                >
                  Dashboard
                </p>
              </a>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-sm-true"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-3"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
            data-shrink={true}
            id="select-filled-label"
          >
            Rating
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="select-filled-label simple-select-filled"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
              id="simple-select-filled"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              9
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value={9}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-sm-true"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-3"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
            data-shrink={false}
            id="mutiple-checkbox-label"
          >
            Genres
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mutiple-checkbox-label mutiple-checkbox"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiInput-input"
              id="mutiple-checkbox"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-root-2"
    >
      <ul
        className="MuiGridList-root makeStyles-gridList-5"
        style={
          Object {
            "margin": -2,
          }
        }
      />
    </div>
  </div>
</div>
`;
